libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/ArrivalProgressObserver.kt

```diff
@@ -52,6 +53,19 @@ internal class ArrivalProgressObserver(
         val routeLegProgress = routeProgress.currentLegProgress
             ?: return
 
+        when (routeProgress.currentState) {
```

Comment:
https://github.com/mapbox/mapbox-navigation-android/blob/bc1dce0cb8cba9ea8c8efc40bd8d65d5350a9d23/libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/ArrivalObserver.kt#L24-L28

nit: could be worth adding docs that mention the onWaypointArrival and onFinalDestinationArrival will only be triggered when the route state is TRACKING or COMPLETE